Plugins/Youbora/YouboraAdnalyzerManager.swift

```diff
@@ -0,0 +1,201 @@
+
+extension YouboraAdnalyzerManager {
+    
+    private var adEventsToRegister: [AdEvent.Type] {
+        return [
+            AdEvent.adLoaded,
+            AdEvent.adStarted,
+            AdEvent.adComplete,
+            AdEvent.adResumed,
+            AdEvent.adPaused,
+            AdEvent.adDidProgressToTime,
+            AdEvent.adSkipped,
+            AdEvent.adStartedBuffering,
+            AdEvent.adPlaybackReady,
+            AdEvent.adsRequested
+        ]
+    }
+    
+    fileprivate func registerAdEvents(onMessageBus messageBus: MessageBus) {
+        PKLog.debug("register ad events")
+        
+        self.adEventsToRegister.forEach { event in
+            PKLog.debug("\(String(describing: type(of: self))) will register event: \(event.self)")
+            
+            switch event {
+            case let e where e.self == AdEvent.adLoaded:
+                messageBus.addObserver(self, events: [e.self]) { [weak self] event in
+                    // update ad info with the new loaded event
+                    self?.adInfo = event.adInfo
+                    // if ad is preroll make sure to call /start event before /adStart
+                    if let positionType = event.adInfo?.positionType, positionType == .preRoll {
+                        self?.plugin.playHandler()
+                    }
+                    self?.playAdHandler()
+                }
+            case let e where e.self == AdEvent.adStarted:
+                messageBus.addObserver(self, events: [e.self]) { [weak self] event in
+                    self?.joinAdHandler()
+                }
+            case let e where e.self == AdEvent.adComplete:
+                messageBus.addObserver(self, events: [e.self]) { [weak self] event in
+                    self?.endedAdHandler()
+                    self?.adInfo = nil
+                }
+            case let e where e.self == AdEvent.adResumed:
+                messageBus.addObserver(self, events: [e.self]) { [weak self] event in
+                    self?.resumeAdHandler()
+                }
+            case let e where e.self == AdEvent.adPaused:
+                messageBus.addObserver(self, events: [e.self]) { [weak self] event in
+                    self?.pauseAdHandler()
+                }
+            case let e where e.self == AdEvent.adDidProgressToTime:
+                messageBus.addObserver(self, events: [e.self]) { [weak self] event in
+                    // update ad playhead with new data
+                    self?.adPlayhead = event.adMediaTime?.doubleValue
+                }
+            case let e where e.self == AdEvent.adSkipped:
+                messageBus.addObserver(self, events: [e.self]) { [weak self] event in
+                    self?.skipAdHandler()
+                }
+            case let e where e.self == AdEvent.adStartedBuffering:
+                messageBus.addObserver(self, events: [e.self]) { [weak self] event in
+                    self?.bufferingAdHandler()
```

Comment:
@gal-orlanczyk need to make sure adsManagerAdPlaybackReady delegate is called also when ad buffers in the middle of the playback and not just initial buffer time, which by the way, AFAIK, should be covered in the join-start time in Youbora, and not by a buffer event.
Please preform the tests to understand this behviour.
Also, `AdPlaybackReady` is not a good event name to indicate ad buffer ended semantically.
Need to make sure we have clear indication for ad buffer start and ad buffer ended.